bikestreets-ios/Map/DefaultMapsViewController.swift

```diff
@@ -232,53 +233,93 @@ final class DefaultMapsViewController: MapsViewController {
       )
     }
   }
-
-  // MARK: - State Handling
-
-  private func requestDirections(request: StateManager.RouteRequest) {
-    let startName: String
-    let endName: String
-
-    switch stateManager.state {
-    case .requestingRoutes(let request):
-      startName = request.origin.name
-      endName = request.destination.name
-    default:
-      startName = "INCORRECT-STATE-FOR-NAME"
-      endName = "INCORRECT-STATE-FOR-NAME"
-    }
-
+  
+  // MARK: - Route Requests
+  private func requestRoute(request: StateManager.RouteRequest) {
     RouteRequester.getOSRMDirections(
-      originName: startName,
       startPoint: request.origin.coordinate,
-      destinationName: endName,
       endPoint: request.destination.coordinate
-    ) { result in
-      switch result {
-      case .success(let result):
-        DispatchQueue.main.async {
-          if let previewRoutes = result.routes {
-            // On initial state update, assume first route is selected.
-            self.stateManager.state = .previewDirections(
-              preview: .init(
-                request: request,
-                response: result,
-                routes: previewRoutes
-              )
+    ) { [weak self] result in
+      DispatchQueue.main.async {
+        self?.handleRouteResponse(result, forRequest: request)
+      }
+    }
+  }
+  
+  private func handleRouteResponse(_ result: Result<CustomRouteResponse, Error>, forRequest request: StateManager.RouteRequest) {
+    switch result {
+    case .success(let result):
+      switch rerouteManager.state {
+      case .rerouting:
+        // Do nothing if a reroute is requested but returns no options
+        guard let _ = result.routes?.first, let router = navigationViewController?.navigationService.router else { break }
```

Comment:
Should this reset back to `rerouteManager.state = .idle` though?
```suggestion
        // Do nothing if a reroute is requested but returns no options
        guard result.routes?.first != nil, let router = navigationViewController?.navigationService.router else {
            rerouteManager.state = .idle
            break 
        }
```